plugins/modules/nd_backup_schedule.py

```diff
@@ -156,35 +157,29 @@ def main():
         )
 
     path = "/api/v1/infra/backups/schedules"
-    schedules = nd.query_obj(path)
-    if name:
-        for schedule in schedules.get("schedules"):
-            if name == schedule.get("name"):
-                nd.previous = nd.existing = schedule
-                path = "{0}/{1}".format(path, name)
-                break
+
+    schedules = nd.get_object_by_nested_key_value(path, "name", name, data_key="schedules")
```

Comment:
This only returns the schedule with the name

Reply:
```
        if not value and response_data and data_key and data_key in response_data:
            return response_data.get(data_key)
        elif not response_data:
            return None
```

I changed the suggested code to support query one and query all objects based on the input value. Should I revert this change?

Reply:
Yes, I think it's best to let the function do what its name suggests